realm/realm-library/src/main/cpp/util.cpp

```diff
@@ -43,195 +43,170 @@ jmethodID session_error_handler;
 
 void ThrowRealmFileException(JNIEnv* env, const std::string& message, realm::RealmFileException::Kind kind);
 
-void ConvertException(JNIEnv* env, const char *file, int line)
-{
-    ostringstream ss;
+jthrowable JThrowableNew(JNIEnv* env, const char* class_name, const char* message) {
+    jclass j_exception_class = env->FindClass(class_name);
+    if (j_exception_class == nullptr) {
+        return nullptr;
+    }
+    jmethodID constructor = env->GetMethodID(j_exception_class, "<init>", "(Ljava/lang/String;)V");
+    if (constructor == nullptr) {
+        env->DeleteLocalRef(j_exception_class);
```

Comment:
no need to call this I think? The local ref will be deleted anyway when the code returns the to java land. But it is minor, i am fine with this.